connectors/sources/salesforce.py

```diff
@@ -169,26 +206,118 @@ async def _select_queryable_fields(self, sobject, fields):
         queryable_fields = sobject_fields.get(sobject, [])
         return [f for f in fields if f.lower() in queryable_fields]
 
-    async def _get_non_bulk_query_pages(self, soql_query):
-        return await self.get_request(
-            QUERY_ENDPOINT,
+    async def _yield_non_bulk_query_pages(self, soql_query):
+        """loops through query response pages and yields lists of records"""
+        response = await self._get_json(
+            f"{self.base_url}{QUERY_ENDPOINT}",
             params={"q": soql_query},
         )
+        yield response.get("records")
+
+        if response.get("done") is False:
+            page_count = 1
+            next_url = response.get("nextRecordsUrl")
+            while True:
+                next_page = await self._get_json(next_url)
+                yield next_page.get("records")
+
+                page_count += 1
+                if (
+                    next_page.get("done") is True
+                    or page_count >= QUERY_PAGINATION_LIMIT
+                ):
+                    break
+                else:
+                    next_url = next_page.get("nextRecordsUrl")
 
     def _token_cleanup(self):
         self.token = None
-        self.token_issued_at = None
-        self.token_refresh_enabled = False
 
     def _auth_headers(self):
         return {"authorization": f"Bearer {self.token}"}
 
-    async def get_request(self, endpoint, params=None):
-        # TODO improve error handling
+    @retryable(
+        retries=RETRIES,
+        interval=RETRY_INTERVAL,
+    )
```

Comment:
I think you'll need custom retryable logic here - no need to retry on invalid query, probably on throttling too